pulp-upload.py

```diff
@@ -0,0 +1,379 @@
+        url = self.url(f"api/v3/content/?pulp_label_select=build_id~{build_id}")
+        return requests.get(url, **self.request_params)
+
+    def get_file_locations(self, artifacts):
+        hrefs = [list(artifact.values())[0] for artifact in artifacts]
+        hrefs_string = ','.join(hrefs)
+        url = self.url(f"api/v3/artifacts/?pulp_href__in={hrefs_string}")
+        return requests.get(url, **self.request_params)
+    
+def upload_rpm(client, rpm_repository_prn, rpm, labels):
+    logging.info(f"Uploading rpm file: {rpm}")
+    content_upload_response = client.create_rpm_content(rpm_repository_prn, rpm, labels)
+    client.wait_for_finished_task(content_upload_response.json()['task'])
+    
+def upload_log(client, file_repository_prn, log, build_id, labels):
+    logging.info(f"Uploading log file: {log}")
+    content_upload_response = client.create_file_content(file_repository_prn, log, build_id, labels)
+    client.wait_for_finished_task(content_upload_response.json()['task'])
+
+    
+def upload_rpms_logs(rpm_path, args, client, arch):
+    rpms = glob.glob(os.path.join(rpm_path,"*.rpm"))
+    logs = glob.glob(os.path.join(rpm_path,"*.log"))
+
+    labels = {
+        "date": f"{now_utc.strftime('%Y-%m-%d %H:%M:%S')}",
+        "build_id": f"{args.build_id}",
+        "arch": f"{arch}",
+        "namespace": f"{args.namespace}",
+        "parent_package": f"{args.parent_package}"
+    }
+    to_await = []
+    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
+        to_await.extend([executor.submit(upload_rpm, client, rpm_repository_prn, rpm, labels) for rpm in rpms])
+        to_await.extend([executor.submit(upload_log, client, file_repository_prn, log, args.build_id,labels) for log in logs])
+        results = [future.result() for future in concurrent.futures.as_completed(to_await)]
+
+def collect_results(client):
+    # Collect results
+    resp_json = client.find_content_by_build_id(args.build_id).json()
+
+    artifacts = [result["artifacts"] for result in resp_json["results"]]
+
+    file_locations_json = client.get_file_locations(artifacts).json()["results"]
+
+    results = {}
+    logging.info(f"Collecting Quay URLs for uploaded files.")
+
+    for artifact in artifacts:
+        for file in file_locations_json:
+            if file["pulp_href"] == list(artifact.values())[0]:
+                results[list(artifact.keys())[0]] = file["file"]
+
+    # write to a results file
+    with open("results.json", "w") as outfile:
+        logging.info(f"Writing Quay URL results to results.json")
+        outfile.write(json.dumps(results, indent = 2))
+
+
+def upload_sbom(client, args):
+    labels = {
+        "date": f"{now_utc.strftime('%Y-%m-%d %H:%M:%S')}",
+        "build_id": f"{args.build_id}",
+        "namespace": f"{args.namespace}",
+        "parent_package": f"{args.parent_package}"
+    }
+    sbom_file = "/var/workdir/results/oras-staging/sbom-merged.json"
```

Comment:
While stuff like `rpm_path` is configurable via CLI option, this SBOM path is hard-coded.
Is this really desired?